### ISSUE TITLE
Coil:Heating:DX:SingleSpeed input processing does not catch blank defrost EIRfT curve 

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5700]
## Issue overview

This object

```
Coil:Heating:DX:SingleSpeed,
    BC Heating Coil System HC,  !- Name
    Always On,               !- Availability Schedule Name
    autosize,                !- Gross Rated Heating Capacity {W}
    3.03,                    !- Gross Rated Heating COP {W/W}
    autosize,                !- Rated Air Flow Rate {m3/s}
    773.3,                   !- Rated Supply Fan Power Per Volume Flow Rate {W/(m3/s)}
    CoilSystem_Cooling_DX 1 DX Cooling Coil System Outlet Node Name,  !- Air Inlet Node Name
    CoilSystem_Heating_DX 1 Air Outlet,  !- Air Outlet Node Name
    HPACHeatCapFT,           !- Heating Capacity Function of Temperature Curve Name
    HPACHeatCapFFF,          !- Heating Capacity Function of Flow Fraction Curve Name
    HPACHeatEIRFT,           !- Energy Input Ratio Function of Temperature Curve Name
    HPACHeatEIRFFF,          !- Energy Input Ratio Function of Flow Fraction Curve Name
    HPACPLFFPLR,             !- Part Load Fraction Correlation Curve Name
    ,                        !- Defrost Energy Input Ratio Function of Temperature Curve Name
    -8.0,                    !- Minimum Outdoor Dry-Bulb Temperature for Compressor Operation {C}
    ,                        !- Outdoor Dry-Bulb Temperature to Turn On Compressor {C}
    5.0,                     !- Maximum Outdoor Dry-Bulb Temperature for Defrost Operation {C}
    0.0,                     !- Crankcase Heater Capacity {W}
    10.0,                    !- Maximum Outdoor Dry-Bulb Temperature for Crankcase Heater Operation {C}
    ,                        !- Defrost Strategy
    ,                        !- Defrost Control
    0.058333,                !- Defrost Time Period Fraction
    autosize,                !- Resistive Defrost Heater Capacity {W}
    4;                       !- Region number for calculating HSPF
```

Fails with this error:
`**  Fatal  ** CurveValue: Invalid curve passed.`

The blank "Defrost Strategy" defaults to ReverseCycle, which requires a valid Defrost Energy Input Ratio Function of Temperature Curve Name, but the input processing does not catch that this is blank and sends a zero curve index to CurveValue.
### Details

Some additional details for this issue (if relevant):
- Version of EnergyPlus Version 8.5.0-3b72813e45
- Helpdesk ticket 11384
### Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Defect file added (DefectFiles\5700)
- [ ] Ticket added to Pivotal for defect (development team task)
- [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by rraustad** [comment]
Well I thought this was correct. It turns out that the defrost EIR curve is needed for ReverseCycle whether Defrost control is set to Timed or OnDemand. In addition, the default defrost strategy is ReverseCycle, so if a user leaves this input blank, the defrost EIR curve is required. Seems a little odd to require a curve object when an input field is blank.

Coil:Heating:DX:SingleSpeed,
    ,                        !- Defrost Energy Input Ratio Function of Temperature Curve Name
    ,                        !- Defrost Strategy
    ,                        !- Defrost Control

Error file now shows:

```
** Severe  ** GetDXCoils: Coil:Heating:DX:SingleSpeed="BC HEATING COIL SYSTEM HC", missing
**   ~~~   ** ...required Defrost Energy Input Ratio Function of Temperature Curve Name is blank.
**   ~~~   ** ...field is required because Defrost Strategy is "ReverseCycle".
**  Fatal  ** GetDXCoils: Errors found in getting Coil:Heating:DX:SingleSpeed input. Preceding condition(s) causes termination.
```

**Comment by mjwitte** [comment]
Closed via #5702.
